### PR TITLE
 modified category module_1

### PR DESCRIPTION
**Pull request by Avetiq30**

### CODE REVIEW

File: src/category/category.controller.ts [hunk 1]
```diff
@@ -27,9 +27,9 @@ export class CategoryController {
     return await this.categoryService.getAllCategories();
   }
 
-  @Get(':name')
+  @Get(':id')
   async getCategoryByName(
```
Comment (VahagShahinyan) [inline src/category/category.controller.ts:31]: Rename ```getCategoryByName``` to ```getCategoryByid```

File: src/category/category.controller.ts [hunk 2]
```diff
@@ -27,9 +27,9 @@ export class CategoryController {
...
-    @Param('name') name: string,
+    @Param('id') name: string,
```
Comment (VahagShahinyan) [inline src/category/category.controller.ts:32]: Change ```@Param('id') name: string,``` to ```@Param('id') id: string```

File: src/category/category.controller.ts [hunk 3]
```diff
@@ -27,9 +27,9 @@ export class CategoryController {
...
   ): Promise<CategoryModel | null> {
     return await this.categoryService.getCategoryByName(name);
```
Comment (VahagShahinyan) [inline src/category/category.controller.ts:34]: Create new service method ```return await this.categoryService.getCategoryById(id);```

File: src/category/category.controller.ts
Comment (VahagShahinyan) [inline src/category/category.controller.ts:1]: Rename 
```updateCategory``` to ```updateCategoryById```
```deleteCategory``` to ```deleteCategoryById```

File: src/category/category.controller.ts (hunk 1 above)
Comment (Avetiq30) [reply src/category/category.controller.ts:31]: ++

File: src/category/category.controller.ts (hunk 2 above)
Comment (Avetiq30) [reply src/category/category.controller.ts:32]: ++

File: src/category/category.controller.ts (hunk 3 above)
Comment (Avetiq30) [reply src/category/category.controller.ts:34]: ++ alternative

File: src/product/product.service.ts [hunk 4]
```diff
@@ -18,9 +18,11 @@ export class ProductService {
   ) {}
 
   async createProduct(createProductDto: ProductCreateDto): Promise<any> {
-    const category = await this.categoryService.getCategoryByName(
+    const category = await this.categoryService.getCategoryByid(
       createProductDto.category,
     );
+    console.log(category);
```
Comment (VahagShahinyan) [inline src/product/product.service.ts:24]: remove ```console.log```

File: src/category/category.controller.ts [hunk 5]
```diff
@@ -8,14 +8,14 @@ import {
...
-import { CreateCategoryDto } from './categoryDto';
+import { CategoryModel } from './model/category.model';
+import { CreateCategoryDto } from './dto/category-create.dto';
 
 @Controller('categories')
 export class CategoryController {
   constructor(private readonly categoryService: CategoryService) {}
 
-  @Post('create')
+  @Post('')
```
Comment (VahagShahinyan) [inline src/category/category.controller.ts:18]: Change ```@Post('')``` to ```@Post()```

File: src/product/product.service.ts (hunk 4 above)
Comment (Avetiq30) [reply src/product/product.service.ts:24]: ++

File: src/category/category.controller.ts (hunk 5 above)
Comment (Avetiq30) [reply src/category/category.controller.ts]: ++

**Review by VahagShahinyan — APPROVED** [review]

**Pull request merged**